web_ui/src/pages/annotator/annotation/list-item-grid.component.tsx

```diff
@@ -98,16 +125,21 @@ const Labels = ({ children, ...otherProps }: FlexProps) => {
 };
 
 const ListMenu = ({ children, ...otherProps }: FlexProps) => {
-    const { isHovered } = useListItemGridContext();
+    const containerRef = useRef<null>(null);
+    const unwrapContainerRef = useUnwrapDOMRef(containerRef);
+    const { isHovered, setIsHovered } = useListItemGridContext();
 
     if (!isHovered) {
         return <></>;
     }
 
     return (
-        <Flex {...otherProps} gridArea={'list-menu'}>
-            {children}
-        </Flex>
+        <>
+            <Flex {...otherProps} gridArea={'list-menu'} ref={containerRef}>
+                {children}
+            </Flex>
+            <InteractOutside containerRef={unwrapContainerRef} onClickOutside={() => setIsHovered(false)} />
```

Comment:
qq: Would `useTransition` work for this? i.e. wrapping `setIsHovered` on a useTransition. We wouldnt need `onClickOutside` and would be cleaner

Reply:
I’m afraid that won’t work, `useTransition` delays execution, but in this case, we need InteractOutside to handle user actions outside the component. For example, when clicking a menu item, it should trigger the menu action and unselect the container. The unselecting part is exactly what `InteractOutside` is meant to handle.